lib/p2p/NodeList.ts

```diff
@@ -100,6 +161,15 @@ class NodeList {
 
     return false;
   }
+
+  private setBanStatus = (node: NodeInstance, status: boolean) => {
+    node.banned = status;
+    return node.save();
+  }
+
+  private nodeIsBanned = (node: NodeInstance) => {
+    return node.reputationScore < NodeList.banThreshold;
```

Comment:
Unless you envision using this check elsewhere in the future, I'd just put the logic directly in the `addReputationEvent` method above. The logic is simple and just a line, so I don't see a need to have a method for it if it's only used once.